presto-tests/src/main/java/com/facebook/presto/tests/DistributedQueryRunner.java

```diff
@@ -517,22 +521,44 @@ else if (coordinatorSidecar) {
         return server;
     }
 
-    private boolean allNodesGloballyVisible()
+    private void waitForAllNodesGloballyVisible(long startTimeInMs)
+            throws Exception
     {
-        int expectedActiveNodesForRm = externalWorkers.size() + servers.size();
-        int expectedActiveNodesForCoordinator = externalWorkers.size() + servers.size();
+        int expectedActiveNodes = externalWorkers.size() + servers.size();
+        Duration timeout = new Duration(100, SECONDS);
 
-        for (TestingPrestoServer server : servers) {
+        for (int serverIndex = 0; serverIndex < servers.size(); ) {
```

Comment:
Does this only check the nodes 1 time per server?

Reply:
I modified it so we check all nodes serially and don't return until all nodes are reporting all other nodes.